nh_clinical/patient.py

```diff
@@ -51,7 +51,34 @@ class NhClinicalPatient(osv.Model):
     ]
 
     @staticmethod
-    def _clean_identifiers(dirty_vals):
+    def _check_identifier_for_bad_chars(value):
+        """
+        Check for bad characters in string
+
+        :param value: string to check
+        """
+        unallowed_chars = '[^a-zA-Z0-9_\-\s]'
+        allowed_chars = re.sub(unallowed_chars, '', value)
+        if allowed_chars != value:
+            raise ValidationError(
+                'Patient identifier can only contain '
+                'alphanumeric characters, hyphens and underscores'
+            )
+
+    @staticmethod
+    def _remove_whitespace(value):
+        """
+        Remove white space from the supplied string, return None if would
+        be empty string
+
+        :param value: string to remove whitespace from
+        :return: string without whitespace
+        """
+        spaces = '\s'
+        val = re.sub(spaces, '', value)
+        return val if val else None
```

Comment:
If we do the above match that borks on _any_ whitespace, then you can get rid of this method and just do `value.strip() ` before `_check_identifier_for_bad_chars()`.